### ISSUE TITLE
Cannot create empty Hive partition for non-bucketed table

### ISSUE THREAD

**Issue by electrum** [prestodb/presto#12002]
The `system.create_empty_partition` procedure fails because the source directory does not exist (no files were written) and thus the rename fails. `SemiTransactionalHiveMetastore.Committer#prepareAddPartition` should probably unconditionally create `currentPath` before checking if it is equal to `targetPath`.

**Comment by electrum** [comment]
cc @wenleix

**Comment by wenleix** [comment]
This is fixed in https://github.com/prestodb/presto/pull/11376/commits/740fcc9169048a99296a2d5eba79003d578cf954, as part of https://github.com/prestodb/presto/pull/11376 (not merged yet).

We can also merge this separately .

cc @jessesleeping

**Comment by jessesleeping** [comment]
Created a separate PR for the single commit. Will rebase the large PR once this is merged.
